From e673b65dec0d2791aaaa522d6931a3030ce39344 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20Mudr=C3=A1k?= <david@moodle.com>
Date: Mon, 7 Sep 2020 15:26:42 +0200
Subject: [PATCH 1/2] Add templates that we want to test

The linting_ok.mustache should not raise any problem. But the other two
use some Ionic3 / Angular specific syntax that would normally make their
linting fail.
---
 local/test/templates/linting_ok.mustache      | 53 +++++++++++++++++++
 .../test/templates/local/mobile/view.mustache | 41 ++++++++++++++
 local/test/templates/mobile_view.mustache     | 41 ++++++++++++++
 3 files changed, 135 insertions(+)
 create mode 100644 local/test/templates/linting_ok.mustache
 create mode 100644 local/test/templates/local/mobile/view.mustache
 create mode 100644 local/test/templates/mobile_view.mustache

diff --git a/local/test/templates/linting_ok.mustache b/local/test/templates/linting_ok.mustache
new file mode 100644
index 0000000000..56f4ad6f1c
--- /dev/null
+++ b/local/test/templates/linting_ok.mustache
@@ -0,0 +1,53 @@
+{{!
+    This file is part of Moodle - http://moodle.org/
+
+    Moodle is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Moodle is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+}}
+{{!
+    @template local_test/linting_ok
+
+    A test for for mustache linting tests.
+
+    Classes required for JS:
+    * none
+
+    Data attributes required for JS:
+    * none
+
+    Context variables required for this template:
+    * message A cleaned string (use clean_text()) to display.
+    * extraclasses Additional classes to apply to the notification.
+    * closebutton Whether a close button should be displayed to dismiss the message.
+    * announce Whether the notification should be announced to screen readers.
+
+    Example context (json):
+    {
+      "message": "Your pants are on fire!",
+      "closebutton": 1,
+      "announce": 1,
+      "extraclasses": "foo bar"
+    }
+}}
+<div class="alert alert-info alert-block fade in {{ extraclasses }}" {{!
+    }}{{# announce }} role="alert"{{/ announce }}{{!
+    }}>
+    {{# closebutton }}<button type="button" class="close" data-dismiss="alert">&times;</button>{{/ closebutton }}
+    {{{ message }}}
+</div>
+{{# js }}
+require(['jquery', 'theme_bootstrapbase/bootstrap'], function($) {
+    // Setup closing of bootstrap alerts.
+    $().alert();
+});
+{{/ js }}
diff --git a/local/test/templates/local/mobile/view.mustache b/local/test/templates/local/mobile/view.mustache
new file mode 100644
index 0000000000..6db063b5f5
--- /dev/null
+++ b/local/test/templates/local/mobile/view.mustache
@@ -0,0 +1,41 @@
+{{!
+    This file is part of Moodle - http://moodle.org/
+
+    Moodle is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Moodle is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+}}
+{{!
+    @template local_test/local/mobile/view
+
+    A test for for mustache linting tests. The file can be set to be excluded from the HTML validation part.
+
+    Classes required for JS:
+    * none
+
+    Data attributes required for JS:
+    * none
+
+    Context variables required for this template:
+    * arg string Argument.
+
+    Example context (json):
+    {
+      "arg": "Value"
+    }
+}}
+{{=<% %>=}}
+<span core-site-plugins-call-ws-on-load
+      name="local_test_method"
+      [params]="{arg: <% arg %>}"
+      [preSets]="{getFromCache: 0, saveToCache: 0}">
+</span>
diff --git a/local/test/templates/mobile_view.mustache b/local/test/templates/mobile_view.mustache
new file mode 100644
index 0000000000..40eaa3e0f2
--- /dev/null
+++ b/local/test/templates/mobile_view.mustache
@@ -0,0 +1,41 @@
+{{!
+    This file is part of Moodle - http://moodle.org/
+
+    Moodle is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Moodle is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+}}
+{{!
+    @template local_test/mobile_view
+
+    A test for for mustache linting tests. The file can be set to be excluded from the HTML validation part.
+
+    Classes required for JS:
+    * none
+
+    Data attributes required for JS:
+    * none
+
+    Context variables required for this template:
+    * arg string Argument.
+
+    Example context (json):
+    {
+      "arg": "Value"
+    }
+}}
+{{=<% %>=}}
+<span core-site-plugins-call-ws-on-load
+      name="local_test_method"
+      [params]="{arg: <% arg %>}"
+      [preSets]="{getFromCache: 0, saveToCache: 0}">
+</span>
-- 
2.26.2

